packages/drivers/routerlicious-driver/src/documentServiceFactory.ts

```diff
@@ -140,7 +170,17 @@ export class RouterliciousDocumentServiceFactory implements IDocumentServiceFact
         // is sufficient to determine if the token will be undefined or not.
         if (token && this.tokenProvider.documentPostCreateCallback !== undefined) {
             try {
-                await this.tokenProvider.documentPostCreateCallback(documentId, token);
+                await PerformanceEvent.timedExecAsync(
+                    logger2,
+                    {
+                        eventName: "DocPostCreateCallback",
+                        docId: documentId,
+                    },
+                    async () => {
+                        if (token && this.tokenProvider.documentPostCreateCallback) {
```

Comment:
This if is unnecessary (duplicate of if block wrapping this whole statement)

Reply:
Yes - kept it there as typescript complains on possible undefined. And if we drop outer if we get some unnecessary telemetry (which maybe ok, as it’s infrequent)